validator/sawtooth_validator/journal/chain.py

```diff
@@ -62,10 +64,28 @@ def __init__(self, consensus,
         }
 
     def _is_block_complete(self, blkw):
+        """
```

Comment:
The completer will only pass block to the journal if they are deem complete. What is your reasoning for "re-checking" the completeness?